toolkit/tools/precacher/precacher.go

```diff
@@ -141,31 +144,56 @@ func getAllRepoData(repoURLs []string, workerTar, buildDir, repoUrlsFile string)
 	defer queryChroot.Close(leaveChrootOnDisk)
 
 	namesToURLs = make(map[string]string)
-	URLList := []string{}
+	allPackageURLs := []string{}
 	for _, repoURL := range repoURLs {
 		// Use the chroot to query each repo for the packages it contains
-		var packageRepoPaths []string
+		var packageRepoURLs []string
 		err = queryChroot.Run(func() (chrootErr error) {
-			packageRepoPaths, chrootErr = getPackageRepoPaths(repoURL)
+			packageRepoURLs, chrootErr = getPackageRepoPathsFromUrl(repoURL)
 			return chrootErr
 		})
 		if err != nil {
 			return nil, err
 		}
+		allPackageURLs = append(allPackageURLs, packageRepoURLs...)
+	}
+
+	// Use the chroot to query each repo for the packages it contains
+	for _, repoFile := range repoFiles {
+		var packageRepoUrls []string
 
-		// We will be searching by the name: "<name>-<version>.<distro>.<arch>", the results from the repoquery will be
-		// in the form of "<PARTIAL_URL>/<name>-<version>.<distro>.<arch>.rpm"
-		for _, packageRepoPath := range packageRepoPaths {
-			packageName := path.Base(packageRepoPath)
-			packageName = strings.TrimSuffix(packageName, ".rpm")
+		// Replace the existing repo file (if it exists) with the one that we want to query
+		destFile := path.Join(chrootRepoDir, path.Base(repoFile))
```

Comment:
Won't you get multiple repo files in the directory if each `repoFile` has a different file name?

Reply:
Ah, correct. Was a mixup of a few fixes. Originally it was always using the same destination file, so they would overwrite each other.

I've moved back to the old flow that places all the files, then calls the query command once.